### PR TITLE
Calculator's basic operations

### PR DESCRIPTION
**Pull request by brenpardal**
### **WHAT'S NEW**
**This update includes:**
*  Calculator's basic operations
*  Error when a number is divided by zero
*  Error when user enter an invalid operation
*  Remove last symbol when "C" is pressed (short click)
*  Clear all when "C" is pressed (long click)

basic_operations

### CODE REVIEW

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 1]
```diff
@@ -1,8 +1,124 @@
 package com.calculatorkt.mvp.model
 
 import com.calculatorkt.mvp.contract.CalculatorContract
-import com.calculatorkt.utils.EMPTY_STRING
+import com.calculatorkt.utils.*
```
Comment (nicolbidone) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:4]: pls avoid using wildcards 
change the setting at
File > Settings > Project Settings > Code Style > Kotlin> Imports > General > Class count to use import with '*'

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 2]
```diff
@@ -1,8 +1,124 @@
...
+                return EMPTY_STRING
+            }
+            else -> {
+                errorEnum = ErrorEnum.ERROR_INVALID_OPERATION
+                return EMPTY_STRING
+            }
+        }
+    }
+
+    override fun getErrorEnum() = errorEnum
```
Comment (SartiMau) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:73]: if you put the variable public you can use the native get and remove this method

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 3]
```diff
@@ -1,8 +1,124 @@
...
+        operator = EMPTY_STRING
+        res = EMPTY_STRING
+        equalsPressed = false
+        partialRes = EMPTY_STRING
+        errorEnum = ErrorEnum.NONE
+    }
+
+    override fun deleteResult(): String {
+        reset()
+        return EMPTY_STRING
```
Comment (SartiMau) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:87]: why does this method need to return something?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt (hunk 1 above)
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt]: Solved. Thanks!

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 4]
```diff
@@ -1,8 +1,124 @@
...
+        if (!firstOperand.isEmpty()) {
+            operator = operatorSymbol
+            setPartial()
+        }
+    }
+
+    override fun getPartialResult() = partialRes
+
+    override fun getResult(): String {
+        if (!firstOperand.isEmpty()) if (operator.isEmpty() && secondOperand.isEmpty()) {
```
Comment (BQuinteros) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:57]: Here you can do return if like you did it with when

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt [hunk 5]
```diff
@@ -1,8 +1,124 @@
...
+    private var equalsPressed = false
+    private var res = EMPTY_STRING
+
+    companion object {
+        private const val BEGIN_INDEX_POSITION = 0
+        private const val LAST_INDEX_POSITION = 1
+    }
+
+    private fun setPartial() {
+        partialRes = firstOperand + operator + secondOperand
```
Comment (BQuinteros) [inline app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:30]: now in kotlin you can do this "$firstOperand$operator$secondOperand" is more efficient

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt (hunk 2 above)
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:80]: Yes, you're right! Solved.

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt (hunk 5 above)
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:30]: Yes! Thanks. Solved.

File: app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt (hunk 3 above)
Comment (brenpardal) [reply app/src/main/java/com/calculatorkt/mvp/model/CalculatorModel.kt:94]: Solved. Deleted method. I pass EMPTY_STRING and use only reset().

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**